src/parsec_client/operations/psa_algorithm.md

```diff
@@ -0,0 +1,347 @@
+
+| `variant` oneof field name | Type     |
+|----------------------------|----------|
+| `hmac`                     | `Hmac`   |
+| `cbc_mac`                  | `CbcMac` |
+| `cmac`                     | `Cmac`   |
+
+###### Hmac message
+
+HMAC algorithm.
+
+| Field name | Type   | Description           |
+|------------|--------|-----------------------|
+| `hash_alg` | `Hash` | Hash algorithm to use |
+
+###### CbcMac message
+
+The CBC-MAC construction over a block cipher. No fields.
+
+**Warning:** CBC-MAC is insecure in many cases. A more secure mode, such as CMAC, is recommended.
+
+###### Cmac message
+
+The CMAC construction over a block cipher. No fields.
+
+##### Truncated message
+
+A truncated MAC algorithm is identical to the corresponding MAC algorithm except that the MAC value
+for the truncated algorithm consists of only the first `mac_length` bytes of the MAC value for the
+untruncated algorithm.
+
+| Field name   | Type         | Description                                  |
+|--------------|--------------|----------------------------------------------|
+| `mac_alg`    | `FullLength` | A MAC algorithm to truncate                  |
+| `mac_length` | `uint32`     | Desired length of the truncated MAC in bytes |
+
+#### Cipher enumeration
+
+Unauthenticated cipher alhorithms.
+
+**Warning:** The unauthenticated cipher API is provided to implement legacy protocols and for use
+cases where the data integrity and authenticity is guaranteed by non-cryptographic means. It is
+recommended that newer protocols use Authenticated Encryption with Associated Data (AEAD).
+
+- `CIPHER_NONE`: this default variant should not be used.
+- `STREAM_CIPHER`: the stream cipher mode of a stream cipher algorithm. The underlying stream cipher
+   is determined by the key type:
+   - To use ChaCha20, use the Chacha20 key type.
+   - To use ARC4, use the Arc4 key type.
+- `CTR`: A stream cipher built using the Counter (CTR) mode of a block cipher. CTR is a stream
+   cipher which is built from a block cipher. The underlying block cipher is determined by the key
+   type. For example, to use AES-128-CTR, use this algorithm with a key of type Aes and a length of
+   128 bits (16 bytes).
+- `CFB`: A stream cipher built using the Cipher Feedback (CFB) mode of a block cipher. The
+   underlying block cipher is determined by the key type.
+- `OFB`: A stream cipher built using the Output Feedback (OFB) mode of a block cipher. The
+   underlying block cipher is determined by the key type.
+- `XTS`: The XTS cipher mode of a block cipher. XTS is a cipher mode which is built from a block
+   cipher. It requires at least one full block of input, but beyond this minimum the input does not
+   need to be a whole number of blocks.
+- `ECB_NO_PADDING`: The Electronic Code Book (ECB) mode of a block cipher, with no padding. The
+   underlying block cipher is determined by the key type. **Warning:** ECB mode does not protect the
```

Comment:
Just to be consistent with the other uses of `**Warning**`, should that part be on a new line? Or is it like this because it's in a list and it would make it look weird?

Reply:
I think it's the later, you can not have list items on multiple lines AFAIK